internal/server/iigo.go

```diff
@@ -35,12 +35,14 @@ func (s *SOMASServer) runIIGOTax() error {
 	clientMap := getNonDeadClients(s.gameState.ClientInfos, s.clientMap)
 	for clientID, v := range clientMap {
 		tax := v.GetTaxContribution()
+		sanction := v.GetSanctionPayment()
 		err := s.takeResources(clientID, tax, "tax")
-		if err == nil {
-			s.gameState.CommonPool += tax
+		err2 := s.takeResources(clientID, sanction, "sanction")
+		if err == nil && err2 == nil {
+			s.gameState.CommonPool += tax + sanction
 			s.clientMap[clientID].TaxTaken(tax)
 		} else {
-			s.logf("Error getting tax from %v: %v", clientID, err)
+			s.logf("Error getting tax or sanctions from %v: %v sanction error: %v", clientID, err, err2)
```

Comment:
I'd handle them one by one right after getting the errs

Reply:
Yeah you're right, changed in latest commit